### ISSUE TITLE
One model of error messages

### ISSUE THREAD

**Issue by luq10** [strapi/strapi#4420]
I integrate my frontend with API builded using Strapi. I want to get error message from API and show it for user. I noticed that errors messages send by Strapi don't have one model.

Examples: 

`message` can be `object[]`
```
{
   "statusCode":400,
   "error":"Bad Request",
   "message":[
      {
         "messages":[
            {
               "id":"Auth.form.error.email.taken",
               "message":"Email is already taken."
            }
         ]
      }
   ]
}
```

`message` can be `object`
```
{
   "statusCode":400,
   "error":"Bad Request",
   "message":{
      "code":"EENVELOPE",
      "command":"API"
   }
}
```

As i see here: https://github.com/strapi/strapi/issues/4249
`message` can be `string`
```
{
  "statusCode":400,
  "error":"Bad Request",
  "message": "Your account has been blocked by the administrator"
}
```

I created a function for get user-friendly error message from this all models, but i think you should decide for one model and send errors always in this model.

**Comment by lauriejim** [comment]
Helo! Can you tell me the request you did please.
I think the first one is an array to let you highlight the input that have an error (in this case email but could also be password)

**Comment by luq10** [comment]
I send `/auth/local/register` request and get this first (`message` is `object[]`) and second (`message` is `object`) response.

**Comment by luq10** [comment]
Next example of error:

`/connect/google/callback?(...)` response me with status code 200 and `error=Grant%3A%20missing%20session%20or%20misconfigured%20provider` as string, not even an json...

**Comment by lauriejim** [comment]
Hello! Thank you for this feedback.
I just created an internal doc that contains all these different responses.

**Comment by luq10** [comment]
Did you planning common format of errors in the future? This should be super useful from the frontend perspective. If not, you should documented this errors format also in external docs for users implementing your REST api in frontend applications.

**Comment by alexandrebodin** [comment]
We sure want to normlize them. Just no time for it right now

**Comment by luq10** [comment]
Awesome!

**Comment by KojoEnch** [comment]
Any progress about this issue please ? I guess we all need to expose error messages in users language, or it's a poor UX.
At the moment I can't find an other way than translating from a "text" key, which is pretty hazardous, since you may change the messages at any time.

As purposed in [that related issue ](https://github.com/strapi/strapi/issues/4249#issuecomment-542590256), could you provide an id in the error.response.data, such as `{ id: 'Auth.form.error.blocked', message: 'Your account has been blocked by an administrator' } `?

Be what it is, kudos for the accomplished work, it's an amazing tool, well documented !

**Comment by lauriejim** [comment]
Hello! Not really for now that's not part of the priorities we have.
We are focusing on other bigger topics, I hope you can understand.
If you want to customize the translations, you can customize the admin files of translations to have your custom ones.
